test/explain/algorithm/test_captum_explainer_heterogeneous.py

```diff
@@ -0,0 +1,215 @@
+        self.relu = torch.nn.ReLU()
+
+    def forward(self, x_dict, edge_index_dict, batch_info=None,
+                edge_label_index=None):
+        x_dict_n = x_dict
+
+        x_dict_n = self.conv(x_dict_n, edge_index_dict)
+
+        x_dict_n = {k: self.relu(v) for k, v in x_dict_n.items()}
+        x_dict_n = self.conv2(x_dict_n, edge_index_dict)
+
+        if (self.model_config.task_level == ModelTaskLevel.graph):
+            x_dict_n = {
+                k: global_add_pool(v, batch_info[k])
+                for k, v in x_dict_n.items()
+            }
+            x = torch.stack(list(x_dict_n.values()), dim=-1).sum(dim=-1)
+        elif (self.model_config.task_level == ModelTaskLevel.edge):
+            assert edge_label_index is not None
+            x = [
+                x_dict_n[node_type][mask]
+                for node_type, mask in edge_label_index.items()
+            ]
+            x = torch.stack(x).sum(dim=0)
+        else:
+            # Node classification
+            x = x_dict_n['paper']
+
+        if self.model_config.mode == ModelMode.binary_classification:
+            if self.model_config.return_type == ModelReturnType.probs:
+                x = x.sigmoid()
+        elif self.model_config.mode == ModelMode.multiclass_classification:
+            if self.model_config.return_type == ModelReturnType.probs:
+                x = x.softmax(dim=-1)
+            elif self.model_config.return_type == ModelReturnType.log_probs:
+                x = x.log_softmax(dim=-1)
+
+        return x
+
+
+def check_hetero_explanation(
+    explanation: HeteroExplanation,
+    node_mask_type: Optional[MaskType],
+    edge_mask_type: Optional[MaskType],
+):
+
+    if node_mask_type == MaskType.attributes:
+        for node_type, node_mask in explanation.node_mask_dict.items():
+            assert node_mask.size() == explanation.x_dict[node_type].size()
+    elif node_mask_type is None:
+        with pytest.raises(KeyError, match="Tried to collect"):
+            _ = explanation.node_mask_dict
+
+    if edge_mask_type == MaskType.object:
+        for edge_type, edge_mask in explanation.edge_mask_dict.items():
+            assert edge_mask.size() == (explanation[edge_type].num_edges, )
+    elif edge_mask_type is None:
+        with pytest.raises(KeyError, match="Tried to collect"):
+            _ = explanation.edge_mask_dict
+
+
+mode_types = [
+    'binary_classification', 'multiclass_classification', 'regression'
+]
+ret_types = ['probs', 'log_probs', 'raw']
+node_mask_types = [MaskType.attributes, None]
+edge_mask_types = [MaskType.object, None]
+task_levels = [ModelTaskLevel.node, ModelTaskLevel.edge, ModelTaskLevel.graph]
+indices = [1, torch.arange(2)]
+
+
+# Note: no reason to believe tests will pass IntegratedGradients
+# but fail for other methods that are supported, so we just test
+# with IntegratedGradients to reduce quantity of tests overall
+@withPackage('captum')
+@pytest.mark.parametrize('method', ['IntegratedGradients'])
```

Comment:
Should we test other methods?

Reply:
All of the other methods are successful (from what I checked), but add significant overhead to the amount of time it takes to test (the Shapley methods can take >10 minutes alone). It's up to you, and it's fairly easy to add the other methods if we'd like to.